atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -91,6 +91,26 @@ public void registerGaugeForTable(Class clazz, String metricName, TableReference
         registerMetric(clazz, metricName, gauge, tag);
     }
 
+    /**
+     * Add a new gauge to a metric (which may already exist)
+     *
+     * This will throw {@link IllegalArgumentException} if a non-Gauge metric has already been added.
```

Comment:
Do we want to handle the possible IllegalArgumentException here? If so, how - catch and log metric registration failure?

Reply:
I'm quite indifferent to the options - the `IllegalArgumentException` would indicate an Atlas/product bug right?

Reply:
Most likely. It would be one that we would want to fix, but wouldn't want to be woken up for - so we should probably `log.warn` instead of throwing. I don't think we actually call this method any more...

Reply:
Oh wait this is the method I added. Yep, we call it, and yep it's only ever used for gauges, so the error should not happen (unless a future bug is written!)